gemfirexd/tools/src/dunit/java/com/pivotal/gemfirexd/transactions/SnapshotTxGIIDUnit.java

```diff
@@ -0,0 +1,399 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package com.pivotal.gemfirexd.transactions;
+
+import java.io.ByteArrayInputStream;
+import java.io.DataInputStream;
+import java.io.File;
+import java.io.Serializable;
+import java.sql.Connection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Properties;
+
+import com.gemstone.gemfire.DataSerializer;
+import com.gemstone.gemfire.cache.*;
+import com.gemstone.gemfire.internal.HeapDataOutputStream;
+import com.gemstone.gemfire.internal.cache.BucketRegion;
+import com.gemstone.gemfire.internal.cache.GemFireCacheImpl;
+import com.gemstone.gemfire.internal.cache.InitialImageOperation;
+import com.gemstone.gemfire.internal.cache.LocalRegion;
+import com.gemstone.gemfire.internal.cache.PartitionedRegion;
+import com.gemstone.gemfire.internal.cache.RegionEntry;
+import com.gemstone.gemfire.internal.cache.versions.RegionVersionVector;
+import com.gemstone.gemfire.internal.shared.Version;
+import com.pivotal.gemfirexd.DistributedSQLTestBase;
+import com.pivotal.gemfirexd.TestUtil;
+import io.snappydata.test.dunit.AsyncInvocation;
+import io.snappydata.test.dunit.SerializableCallable;
+import io.snappydata.test.dunit.SerializableRunnable;
+import io.snappydata.test.dunit.VM;
+
+
+public class SnapshotTxGIIDUnit extends DistributedSQLTestBase {
+
+  public SnapshotTxGIIDUnit(String name) {
+    super(name);
+  }
+
+  public static final String regionName = "GII_TEST_T1";
+
+  @Override
+  protected String reduceLogging() {
+    return "fine";
+  }
+
+  public static File[] getDiskDirs() {
+    return new File[]{getDiskDir()};
+  }
+
+  private static File getDiskDir() {
+    int vmNum = VM.getCurrentVMNum();
+    File dir = new File("diskDir", "disk" + String.valueOf(vmNum)).getAbsoluteFile();
+    dir.mkdirs();
+    return dir;
+  }
+
+  @Override
+  public void setUp() throws Exception {
+    System.setProperty("gemfire.cache.ENABLE_DEFAULT_SNAPSHOT_ISOLATION", "true");
+    invokeInEveryVM(new SerializableRunnable() {
+      @Override
+      public void run() {
+        System.setProperty("gemfire.cache.ENABLE_DEFAULT_SNAPSHOT_ISOLATION", "true");
+        System.setProperty("snappydata.snapshot.isolation.gii.lock", "true");
+      }
+    });
+    super.setUp();
+  }
+
+  @Override
+  public void tearDown2() throws Exception {
+    System.setProperty("gemfire.cache.ENABLE_DEFAULT_SNAPSHOT_ISOLATION", "false");
+    invokeInEveryVM(new SerializableRunnable() {
+      @Override
+      public void run() {
+        System.clearProperty("gemfire.cache.ENABLE_DEFAULT_SNAPSHOT_ISOLATION");
+        System.clearProperty("snappydata.snapshot.isolation.gii.lock");
+      }
+    });
+    super.tearDown2();
```

Comment:
I think super.tearDown should always be called in try...finally...Here it won't matter I guess as the code here is unlikely to fail but I guess we need that cleanup at any cost hence the suggestion.